sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -131,38 +146,34 @@ void setPackageDeclaration(String packageName) {
     void createClass(String serviceInterfaceImplShortName, String serviceInterfaceShortName,
         TemplateInput templateInput) {
         classBuilder = compilationUnit.addClass(serviceInterfaceImplShortName, Modifier.Keyword.PUBLIC);
-
         String serviceInterfacePackage = templateInput.getServiceInterfaceFQN()
             .substring(0, templateInput.getServiceInterfaceFQN().lastIndexOf('.'));
+
         compilationUnit.addImport(serviceInterfacePackage + "." + serviceInterfaceShortName);
         classBuilder.addImplementedType(serviceInterfaceShortName);
 
         configureLoggerField(classBuilder.addField("ClientLogger", "LOGGER", Modifier.Keyword.PRIVATE,
             Modifier.Keyword.STATIC, Modifier.Keyword.FINAL), serviceInterfaceShortName);
 
-        classBuilder.addField(HttpPipeline.class, "defaultPipeline", Modifier.Keyword.PRIVATE, Modifier.Keyword.FINAL);
-        classBuilder.addField(ObjectSerializer.class, "serializer", Modifier.Keyword.PRIVATE, Modifier.Keyword.FINAL);
-
+        classBuilder.addField(HttpPipeline.class, "httpPipeline", Modifier.Keyword.PRIVATE, Modifier.Keyword.FINAL);
         compilationUnit.addImport(JsonSerializer.class);
+        classBuilder.addField(JsonSerializer.class, "jsonSerializer", Modifier.Keyword.PRIVATE, Modifier.Keyword.FINAL);
+        compilationUnit.addImport(XmlSerializer.class);
+        classBuilder.addField(XmlSerializer.class, "xmlSerializer", Modifier.Keyword.PRIVATE, Modifier.Keyword.FINAL);
+
         classBuilder.addConstructor(Modifier.Keyword.PRIVATE)
-            .addParameter(HttpPipeline.class, "defaultPipeline")
-            .addParameter(ObjectSerializer.class, "serializer")
+            .addParameter(HttpPipeline.class, "httpPipeline")
             .setBody(StaticJavaParser.parseBlock(
-                "{ this.defaultPipeline = defaultPipeline; this.serializer = serializer == null ? new JsonSerializer() : serializer; }"));
+                "{ this.httpPipeline = httpPipeline; this.jsonSerializer = new JsonSerializer(); this.xmlSerializer = new XmlSerializer(); }"));
 
         classBuilder.addMethod("getNewInstance", Modifier.Keyword.PUBLIC, Modifier.Keyword.STATIC)
             .setType(serviceInterfaceShortName)
             .addParameter(HttpPipeline.class, "pipeline")
-            .addParameter(ObjectSerializer.class, "serializer")
-            .setBody(StaticJavaParser
-                .parseBlock("{ return new " + serviceInterfaceImplShortName + "(pipeline, serializer); }"))
+            .setBody(StaticJavaParser.parseBlock("{ return new " + serviceInterfaceImplShortName + "(pipeline); }"))
```

Comment:
Probably want to be consistent with using `pipeline` or `httpPipeline` as the field / variable name. Will make it easier to do refactorings in the future if needed.